### ISSUE TITLE
Localhost is ok, activities before to work in cluster.

### ISSUE THREAD

**Issue by andresvivancov** [andresvivancov/BDAPRO-WindowsBenchmarking#24]
Dear @jeyhunkarimov, my approach and task before to work in the IBM Power Cluster is this:

![image](https://cloud.githubusercontent.com/assets/15791882/21472324/4ea69ee4-cad4-11e6-87d8-d94b464700e5.png)

I am working with 2 virtual machines, for difference the creation of datastream with flink and kafka with eventime, in the other virtual machine I am reading this event and adding windows for the calculation of latency.  Is it okay to advance in it before to try in the IBM Cluster?

**Comment by jeyhunkarimov** [comment]
@andresvivancov once you calculate the latency as we discussed in the meeting, the design seems to be ok. Please, do the experiments as indicated in the original description of the project, and upload create a document with figures and respective explanations.

**Comment by andresvivancov** [comment]
@jeyhunkarimov thank you, I am on it.

**Comment by andresvivancov** [comment]
Hello @jeyhunkarimov , I created 3 nodes Kafka (in 3 virtual machines separately ) also Zookeepers and kafka brokers are co-located on the same nodes., and it is working, If I need I can create more virtual machines If I need. I have good performance with 64GB in ram.

My question is could I anticipate my project in my own cluster and maybe do not use the IBM Power? or is indispensable to work in the IBM Power Cluster?

Thank you 

Andres

A screenshot of it:

![image](https://cloud.githubusercontent.com/assets/15791882/21510075/aa8d9ef8-cc8f-11e6-884e-aa6686b342fd.png)

**Comment by andresvivancov** [comment]
@jeyhunkarimov  yes its true sorry, I was a bit worried for the message that I received when I asked credential for the cluster that said **"No change the Clustersoftware!"**,   but its okay I will start working in IBM Power Cluster. Thank you for your support 

Andres

**Comment by andresvivancov** [comment]
Hello @jeyhunkarimov , I wanted to install Kafka, Zookeeper in IBM Cluster. and I supposed to work for zookeeper in  dataDir=/var/zookeeper/data and for kafka logs in /tmp/kafka-logs/. 

But in the chat of the IBM Cluster I asked about it and Alexander Alexandrov told me that could be better talk with you for any suggestion to decide where to write the partitions. based and what we told and my researching the best solution could be:

Installation of Kafka /share/hadoop/kafka
Zookeeper Data in: /share/hadoop/kafka/data/zookeeper/data
Kafka Logs in:  ~/tmp/kafka-logs/

What are your comments, please?

Thanks

**Comment by jeyhunkarimov** [comment]
@andresvivancov, 
1. Did you finished the local version and tested? If yes can I see the results?
2. Zookeeper/kafka dont need installation . They are just jars. For further information, you can check related documentation of zookeeper/kafka

**Comment by andresvivancov** [comment]
@jeyhunkarimov 
1. Yes I finished it, its running without error, as you said before I need to work and run the experiments in the IBM Cluster, so I was preparing the ecosystem for doing the experiments and do the report there with the official results. But if you like a previous version of the experiments in my own cluster I can do it tomorrow.
2. I installed / or copied the jars or installation folder in my own cluster, and for it , I need to configure some paths for zookeeper and for kafka-logs.

**Comment by jeyhunkarimov** [comment]
@andresvivancov, 
In the last meeting I think we aggreed that you run experiments locally and send results in figures to me, then you will continue on cluster. 
Nevertheless, if you want to get results directly  on cluster, thats ok. Just ping me once you get results on cluster environment.

**Comment by andresvivancov** [comment]
@jeyhunkarimov I agree with you Jeyhun we aggreed in something but in github your approach is "hi @andresvivancov. As we mentioned in the class several times, the main purpose of this project is getting experience with cluster environment...." so I was confused that the new approach is focus on IBM Cluster. Sorry, Okay I will work  first in a report in a local environment, and I will send you. Is it okay for you?

**Comment by jeyhunkarimov** [comment]
@andresvivancov
The reason that we first do experiments on locally is that it is easy to solve peoblems. After conducting locally, we do experiments on cluster. So, doing locally is for yourself, it will not be counted (or counted with little impact) to your grade. Thats why I mentioned this. However, if you want to go to endpoint directly, you can directly do experiments in cluster of course.

**Comment by andresvivancov** [comment]
Dear @jeyhunkarimov 

I understand you point , yes you have reason, to be honest I have running multi node kafka as we agreed separate in different virtual machines . I had many many problems but now it is fixed everything. It was not easy but now is working.
Hence I will prepare a report with the experiments in local for you and I can based on it for final experiments. I will do it tomorrow. 

But please also I need to work later on the IBM Cluster, and Alexander wrote : _please check with your advisor (if you're doing this for a thesis) or course leader (if you're doing this for a course project) if you need any assistance in setting up these systems_

So I need your help or confirmation for where do I need to write the partitions that I mention before? I do not want to collapse the server or something. Could you help me with it please?

I based on the official version for the configuration : https://kafka.apache.org/quickstart ; and the solution could be: 

Installation of Kafka /share/hadoop/kafka
Zookeeper Data in: /share/hadoop/kafka/data/zookeeper/data
Kafka Logs in: ~/tmp/kafka-logs/

Thank you Jeyhun

Kind regards

Andres

**Comment by jeyhunkarimov** [comment]
@andresvivancov just create one folder and initialize all the applications within that folder (in different subfolders). In  this way, you will not intervene with other services.

**Comment by andresvivancov** [comment]
@jeyhunkarimov good afternoon,  perfect! thank you so much, :) I was talking with Patrick for working together in the IBM Cluster especially for the configurations of Kafka. I am still working in the report with the experiments in local for you can appreciate better in advance the design of the benchmark and charts. 
Have a nice day.
Thanks
